code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/file/property/OpendPropertyFiles.xtend

```diff
@@ -0,0 +1,101 @@
+package de.fhdo.lemma.model_processing.code_generation.container_base.file.property
+
+import java.util.HashSet
+import java.util.Set
+import java.io.File
+
+/**
+ * This class manages several different application.properties files, used in the spring framework.
+ * The class is implemented by the usage of the singleton design pattern.
+ *
+ *
+ * @author [Philip Wizenty](mailto:philip.wizenty@fh-dortmund.de)
+ */
+class OpendPropertyFiles {
+    static OpendPropertyFiles instance
+    val Set<PropertyFile> propertyFiles = new HashSet
+
+    static def OpendPropertyFiles getInstance() {
+        if (OpendPropertyFiles.instance === null)
+            OpendPropertyFiles.instance = new OpendPropertyFiles
+
+        return OpendPropertyFiles.instance
+    }
+
+    /*
+     * Add the given [propertyFile] to the store or replace it, if it already exists
+     */
+    def addOrReplace(PropertyFile propertyFile) {
+        propertyFiles.removeIf(p | p.filePath == propertyFile.filePath)
+        propertyFiles.add(propertyFile)
+    }
+
+    /*
+     * Get the [PropertyFile] with the given [filePath] from the store
+     */
+    def PropertyFile get(String propertyName) {
+        propertyFiles.findFirst[it.filePath == filePath]
+    }
+
+    /*
+     * Open the property file at the given [filePath]. If the path does not represent an existing
+     * property file, a new [PropertyFile] instance is created. In any case, the [PropertyFile]
+     * instance is stored in the [OpenedPropertyFiles]. These instances are reused upon calling this
+     * function, i.e., if the [OpenedPropertyFiles] contains an instance for the given [filePath] it
+     * is returned by the function.
+     */
+    def PropertyFile openPropertyFile(String filePath) {
+        var propertyFile = propertyFiles.findFirst[f | f.filePath == filePath]
+
+        if (propertyFile !== null) {
```

Comment:
I see several improvements here:
    - if (like switch) is an Xtend expression. You may therefore pull-out the return.
    - unnecessary brackets
    - The File object is created twice and File.length will return 0 even for non-existing files.

Proposal:

```
return switch (propertyFile) {
    case propertyFile !== null: propertyFile
    case new File(filePath).length > 0: PropertyFile.loadFromFilePath(filePath)
    default: new PropertyFile(filePath, new SortableProperties)
}
```

Another possible solution:
Since `loadFromFilePath` is static in `PropertyFile` let it decide itself what
to do with the `filePath` and act like a factory method if possible, i.e.,

```
static def PropertyFile loadFromFilePath(String filePath) {
    val file = new File(filePath)
    if (file.length == 0)
        return new PropertyFile(filePath, new SortableProperties)

    val loadedPropertiers = new SortableProperties()
    val reader = new BufferedReader(new FileReader(file));
    loadedPropertiers.load(reader)
    return new PropertyFile(filePath, loadedPropertiers)
}
```

Then the previous if-statement would become even easier:
```
return if (propertyFile !== null)
        propertyFile
    else
        PropertyFile.loadFromFilePath(filePath)
```